lib/watir/locators/element/locator.rb

```diff
@@ -70,39 +71,58 @@ def using_selenium(filter = :first)
 
         def using_watir(filter = :first)
           selector = selector_builder.normalized_selector
-          visible = selector.delete(:visible)
-          visible_text = selector.delete(:visible_text)
           tag_name = selector[:tag_name].is_a?(::Symbol) ? selector[:tag_name].to_s : selector[:tag_name]
           validation_required = (selector.key?(:css) || selector.key?(:xpath)) && tag_name
 
           if selector.key?(:index) && filter == :all
             raise ArgumentError, "can't locate all elements by :index"
```

Comment:
What do you think about moving this to line 160 to avoid the extra conditionals here?

Reply:
I assume you are okay with the performance hit - ie we'd go grab elements from the page, only to later say we cannot do anything?

Given this should only happen in very rare occurrences, it seems like a reasonable change. Though sadly it doesn't seem to just be a matter of moving the line of code. I got a failed spec when trying, so I'll have to take another stab at it.

Reply:
It'll only be a performance hit if the test is going to fail, which I don't care about optimizing. If it isn't trivial though don't worry about it